Payment gateway

utkhagni13 opened the pull request.

utkhagni13 committed:
adding search box code

utkhagni13 committed:
adding searching

utkhagni13 committed:
adding searching

subodhr258 committed:
feat:ProductCard Rating

subodhr258 committed:
Update User Profile Branch

subodhr258 committed:
feat:Edit

subodhr258 committed:
Made Changes

subodhr258 committed:
deleted extra UserDetails file

Ashes-Mondal committed:
userDetails state error fixed

Ashes-Mondal committed:
Merge branch 'cart-branch' of https://github.com/Ashes-Mondal/Ecommerce-Website into Search-Box

subodhr258 committed:
Merge branch 'user-profile' of https://github.com/Ashes-Mondal/Ecommerce-Website into cart-branch

subodhr258 committed:
fixed a bug

subodhr258 committed:
solved merged conflict

subodhr258 committed:
Merge conflict

subodhr258 committed:
feat: Edit User #1

Ashes-Mondal committed:
search-box made functional

Ashes-Mondal committed:
search functionality

Ashes-Mondal committed:
search functionality improved

utkhagni13 committed:
merge conflict resolve

utkhagni13 committed:
conflict resolved

subodhr258 committed:
feat:deleteUser

subodhr258 committed:
deleted .env

Ashes-Mondal committed:
user-profile changes

Ashes-Mondal committed:
resolved incoming conflicts

Ashes-Mondal committed:
fixed bug

utkhagni13 committed:
accept changes

utkhagni13 committed:
accept new changes

utkhagni13 committed:
accept new changes

utkhagni13 committed:
accept new changes

subodhr258 committed:
feat: Add Orders

subodhr258 committed:
Resolved incoming conflicts

Ashes-Mondal committed:
admin added to user schema

Ashes-Mondal committed:
resolved merge conflict

Ashes-Mondal committed:
admin route added

utkhagni13 committed:
accept new changes

Ashes-Mondal committed:
fixed session timeout redering

utkhagni13 committed:
accept new changes

Ashes-Mondal committed:
admin and user route merged

subodhr258 committed:
feat: Completed Order functionality

utkhagni13 committed:
accept new changes

Ashes-Mondal committed:
merged remote user-profile

utkhagni13 committed:
accept new changes

utkhagni13 committed:
accept new changes

utkhagni13 committed:
accept new changes

utkhagni13 committed:
accept new changes

utkhagni13 committed:
accept new changes

subodhr258 committed:
Made changes

Ashes-Mondal committed:
merged user-profile-2

utkhagni13 committed:
accept new changes

Ashes-Mondal committed:
userDetails changed

subodhr258 committed:
fix: orders bug fixed

subodhr258 committed:
resolved conflict

subodhr258 committed:
fixed a bug

subodhr258 committed:
feat:name on navbar

subodhr258 committed:
removed Clear Order History button

subodhr258 committed:
Made changes in appearance

utkhagni13 committed:
changes done

utkhagni13 committed:
changes done

utkhagni13 committed:
accept new changes

utkhagni13 committed:
changes done

utkhagni13 committed:
changes done

utkhagni13 committed:
changes done

utkhagni13 committed:
changes done

subodhr258 committed:
Improved Appearance

utkhagni13 committed:
payment gateway

subodhr258 committed:
Merge branch 'payment-gateway' of https://github.com/Ashes-Mondal/Ecommerce-Website into payment-gateway

subodhr258 committed:
made changes

subodhr258 committed:
fixed orderId bug

subodhr258 committed:
Admin page side navbar

Ashes-Mondal committed:
admin routes

Ashes-Mondal committed:
minute changes

subodhr258 committed:
feat:added React Loader

Ashes-Mondal committed:
merged changes

subodhr258 committed:
Added Floating Labels in Signup

Ashes-Mondal committed:
integration

utkhagni13 committed:
accept changes

utkhagni13 committed:
accept changes

utkhagni13 committed:
accept changes

utkhagni13 committed:
accept changes

utkhagni13 committed:
accept changes

subodhr258 committed:
Merge branch 'payment-gateway' of https://github.com/Ashes-Mondal/Ecommerce-Website into admin-dashboard

subodhr258 committed:
Items in Admin

utkhagni13 committed:
accept changes

utkhagni13 committed:
Merge branch 'admin-dashboard' of https://github.com/Ashes-Mondal/Ecommerce-Website into admin-dashboard

Ashes-Mondal committed:
Users_admin-dashboard

Ashes-Mondal committed:
admin-users-page made funtional

subodhr258 committed:
Fixed side navbar

subodhr258 committed:
Merge branch 'admin-dashboard' of https://github.com/Ashes-Mondal/Ecommerce-Website into admin-dashboard

utkhagni13 committed:
accept changes

utkhagni13 committed:
pulling new changes

utkhagni13 committed:
pulling new changes

Ashes-Mondal committed:
admin Order page made

Ashes-Mondal committed:
Merge branch 'admin-dashboard' of https://github.com/Ashes-Mondal/Ecommerce-Website into integration

subodhr258 committed:
feat: Edit Items

utkhagni13 committed:
made new changes

utkhagni13 committed:
made new changes

utkhagni13 committed:
made new changes

Ashes-Mondal committed:
Orders page

subodhr258 committed:
Added css styling

Ashes-Mondal committed:
admin order page

Ashes-Mondal committed:
admin order page

subodhr258 committed:
Delivery Status display in Order Details

utkhagni13 committed:
error removed

utkhagni13 committed:
accept new changes

utkhagni13 committed:
accept new changes

subodhr258 committed:
Added CSS styling

Ashes-Mondal committed:
login validation check

Ashes-Mondal committed:
Merge branch 'admin-dashboard' of https://github.com/Ashes-Mondal/Ecommerce-Website into integration

abhishoya commented on api/search.js:
```diff
@@ -0,0 +1,52 @@
+
+const filterSearch = (itemList, Search) => {
+  for (let wordIndex = 1; wordIndex < Search.length; wordIndex++) {
+    console.log("itemList:", itemList);
```
remove console log statements throughout the code

927tanmay commented on client/src/App.js:
```diff
@@ -27,61 +31,109 @@ const App = () => {
+      //fetching user data from the server side if any
       const userData = await (await fetch("/getUserDetails")).json();
       console.log("userData:", userData);
+      if (userData.response == true || userData.error == "not logged in") {
```
try to use "==="

abhishoya commented on client/src/App.js:
```diff
@@ -27,61 +31,109 @@ const App = () => {
-    const fetchJSON_fromServer = async () => {
-      const itemList = await (await fetch(URL)).json();
-      dispatch({ type: "fetchedItemJson", payload: itemList });
+    const fetchItems_fetchUser = async () => {
```
use camel case for variable name

abhishoya commented on client/src/App.js:
```diff
@@ -27,61 +31,109 @@ const App = () => {
-      dispatch({ type: "fetchedItemJson", payload: itemList });
+    const fetchItems_fetchUser = async () => {
+      //fetching user data from the server side if any
       const userData = await (await fetch("/getUserDetails")).json();
```
don't use nested await. Either use await in different vars or use then-catch statements

abhishoya commented on client/src/Components/Card.js:
```diff
@@ -27,15 +28,16 @@ const Card = (props) => {
     try {
-      const response = await (await fetch("/addToCart", requestOptions)).json();
-      if (response) {
+      const result = await (await fetch("/addToCart", requestOptions)).json();
```
don't use nested await. Either use await in different vars or use then-catch statements

abhishoya commented on client/src/Components/SearchBox.js:
```diff
@@ -1,13 +1,51 @@
 
-const SearchBox = (props) => {
+const SearchBox = () => {
+  const [Search, setSearch] = useState("");
```
s should be small

abhishoya commented on client/src/Components/SearchBox.js:
```diff
@@ -1,13 +1,51 @@
+      headers: { "Content-Type": "application/json" },
+      body: JSON.stringify({ Search: Search }),
+    };
+    const result = await (await fetch("/search", requestOptions)).json();
```
don't use nested await. Either use await in different vars or use then-catch statements

abhishoya commented on client/src/Components/pages/Login.js:
```diff
@@ -1,12 +1,15 @@
   return (
     <div>
-      <form className="box shadow-lg mt-5" action="/login" method="POST">
+      {/*className="login-form"*/}
```
remove JSX comments

abhishoya commented on client/src/Components/pages/ShoppingCart.js:
```diff
@@ -54,7 +62,6 @@ function ShoppingCart({ cart, setCart }) {
       body: JSON.stringify({
-        userId: user._id,
         itemId: Product.item._id,
         itemQty: Product.Qty - 1,
```
product.Qty can be product.qty if it's a minor change

927tanmay commented on client/src/Components/pages/ShoppingCart.js:
```diff
@@ -80,29 +90,159 @@ function ShoppingCart({ cart, setCart }) {
+        }
+      },
+      prefill: {
+        email: "ashes@Rajpopat.com",
```
Will the email and phone no. be hardcoded in prefill?

927tanmay commented on client/src/Components/pages/ShoppingCart.js:
```diff
@@ -146,12 +286,42 @@ function ShoppingCart({ cart, setCart }) {
+              </Link>{" "}
+            </button>
+          ) : (
+            <></>
```
Either use null or like orderHistory && ( ...............)

abhishoya commented on client/src/Components/pages/UserDetails.js:
```diff
@@ -1,52 +1,235 @@
-          </h6>
-          <input type="text" value={user.email} name="Email" disabled />
+    <>
+      <div className="product flex-container userbackground">
```
userbackground can be like user-background. Try follow snake case for CSS classes

abhishoya commented on client/src/Components/pages/UserDetails.js:
```diff
@@ -1,52 +1,235 @@
+                        <b>Order ID : </b>
+                        {element._id}
+                      </p>
+                      {element.order.map((item, i) => {
```
item doesn't make sense here. can use a more semantic variable name

subodhr258 committed:
feat: rating

abhishoya commented on models/userScema.js:
```diff
@@ -29,6 +31,7 @@ const userDetailSchema = mongoose.Schema({
   password: {
     type: String,
     required: true,
+    unique: true,
```
why is password unique true?

abhishoya reviewed (changes requested):
The comments maybe at only one place, but if you see a similar mistake somewhere else too, do rectify it.

subodhr258 committed:
resolving conflict

subodhr258 committed:
resolving conflict

subodhr258 committed:
Fixed a bug

utkhagni13 committed:
making dashboard

utkhagni13 committed:
making dashboard

utkhagni13 committed:
making dashboard

utkhagni13 committed:
making dashboard

utkhagni13 committed:
making dashboard

Ashes-Mondal committed:
fixed issue of reloading

Ashes-Mondal committed:
Merge branch 'admin-dashboard' of https://github.com/Ashes-Mondal/Ecommerce-Website into integration-2

Ashes-Mondal committed:
Search-box brought to life back

subodhr258 committed:
Resolved Warnings

subodhr258 committed:
resolved conflicts

Ashes-Mondal committed:
redirect added to admin route

subodhr258 committed:
Fix: Fixed the ItemDetails Undefined Bug. Resolved some warnings.

subodhr258 committed:
Merge branch 'admin-dashboard' of https://github.com/Ashes-Mondal/Ecommerce-Website into admin-dashboard

Ashes-Mondal committed:
feat-Modal and alerts applied

Ashes-Mondal committed:
issue fixed:user now can't delete account when they have pending orders

subodhr258 committed:
feat: Sorting and Rating

subodhr258 committed:
Resolved conflicts

Ashes-Mondal committed:
Multiple Changes:
1.Login ,Signup UI changed(MUI)
2.Admin order page and Admin user page changed(MUI)

Ashes-Mondal committed:
Resolved merge conflict

Ashes-Mondal committed:
fixed:HomePage container css

Ashes-Mondal committed:
fixed:search button made functional onClick

utkhagni13 committed:
feat: Contact Us Page

utkhagni13 committed:
feat: Ratings

utkhagni13 committed:
feat: Ratings

utkhagni13 committed:
feat: Ratings

Ashes-Mondal committed:
changed:Homepage search button and bootstrap grids is been used in the homepage

Ashes-Mondal committed:
feat:Backend forgot password functions implemented with nodemailer

utkhagni13 committed:
making few changes

utkhagni13 committed:
Merge branch 'admin-dashboard' of https://github.com/Ashes-Mondal/Ecommerce-Website into admin-dashboard

Ashes-Mondal committed:
feat-cards UI

utkhagni13 committed:
changes in contact page

utkhagni13 committed:
accepting new changes

Ashes-Mondal committed:
issue:trying to make responsive cards

subodhr258 committed:
HomePage: UI

utkhagni13 committed:
remove sticky top

utkhagni13 committed:
accepting new changes

subodhr258 committed:
Home Page + Product Details: version 1

subodhr258 committed:
HomePage + ProductDetail version 1.1, comments removed

utkhagni13 committed:
made forgot password

utkhagni13 committed:
Merge branch 'UI' of https://github.com/Ashes-Mondal/Ecommerce-Website into UI

Ashes-Mondal committed:
Resolved merged conflict

Ashes-Mondal committed:
feat:admin can now make others admin and also can revoke the power

Ashes-Mondal committed:
resolved merged conflict

Ashes-Mondal committed:
issue:automated admin dashboard delivered order count

Ashes-Mondal committed:
feat:Forgot password with otp added

Ashes-Mondal committed:
feat:order,user page divided into tabs and Search bar is added to admin pages
